util/metadata_test.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import (
+	"compress/gzip"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestParseMetadataFromGZip_Success(t *testing.T) {
+	f, _ := os.Open("gzip_testfile.tar.gz")
```

Comment:
```suggestion
	f, _ := os.Open("gzip_testfile.tar.gz")
	defer f.Close()
```

Free the OS file handler.